<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="arm">
    <link name = "axis">
    <visual>
        <origin xyz="0 0 0" rpy="0 0 0" />
    </visual>
    </link>

    <link name="base_link">
    <visual>
        <origin xyz="0 0 0" rpy="1.57 0 0" />
        <geometry>
          <mesh filename="package://robot_arm/meshes/base.stl" scale="0.01 0.01 0.01"/>
        </geometry>
        <material name="grey">
        <color rgba="0.83 0.83 0.83 1.0"/>
      </material>
    </visual>
    <collision>
    </collision>
    </link>

  <joint name="axis_to_base" type="fixed">
    <origin xyz="0 0 0" rpy="0 0 0"/> 
    <parent link="axis"/>
    <child link="base_link"/>
  </joint>

  <link name="link1">
    <visual>
     <origin xyz="0 0 0" rpy="1.57 0 1.57" />
        <geometry>  
         <mesh filename="package://robot_arm/meshes/link1.stl" scale="0.01 0.01 0.01"/>
        </geometry>
         <material name="grey">
        <color rgba="0.83 0.83 0.83 1.0"/>
      </material>
    </visual>
    <inertial>
      <mass value="10"/>
      <inertia ixx="1e-3" ixy="0.0" ixz="0.0" iyy="1e-3" iyz="0.0" izz="1e-3"/>
    </inertial>
  </link>

   <joint name="base_to_link1" type="revolute">
   <axis xyz="0 0 1"/>
   <origin xyz="0 0 0.55" rpy="0 0 0"/>
   <limit effort="10" velocity="1" lower="-3.14" upper="3.14"/>
    <parent link="base_link"/>
    <child link="link1"/>
  </joint>

 
  <link name="link2">
    <visual>
        <origin xyz="1 -0.23 0.4" rpy="-1.57 0 1.57" />
        <geometry>
         <mesh filename="package://robot_arm/meshes/link2.stl" scale="0.01 0.01 0.01"/>
        </geometry>
         <material name="grey">
        <color rgba="0.83 0.83 0.83 1.0"/>
      </material>
    </visual>
    <collision>
    </collision>
    <inertial>
      <mass value="10"/>
      <inertia ixx="1e-3" ixy="0.0" ixz="0.0" iyy="1e-3" iyz="0.0" izz="1e-3"/>
    </inertial>
  </link>

   <joint name="link1_to_link2" type="revolute">
   <axis xyz="0 1 0"/>
   <origin xyz="0 0 0" rpy="0 0 0"/>
   <limit effort="10" velocity="1" lower="-3.14" upper="3.14"/>
    <parent link="link1"/>
    <child link="link2"/>
  </joint>

  <link name="link3">
    <visual>
        <origin xyz="0.34 -0.03 0" rpy="-1.57 0 1.57" />
        <geometry>
        <mesh filename="package://robot_arm/meshes/link3.stl" scale="0.01 0.01 0.01"/>
        </geometry>
         <material name="grey">
        <color rgba="0.83 0.83 0.83 1.0"/>
      </material>
    </visual>
    <collision>
       <geometry>
        <mesh filename="package://robot_arm/meshes/link3.stl" scale="0.01 0.01 0.01"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="10"/>
      <inertia ixx="1e-3" ixy="0.0" ixz="0.0" iyy="1e-3" iyz="0.0" izz="1e-3"/>
    </inertial>
  </link>

   <joint name="link2_to_link3" type="revolute">
   <axis xyz="0 1 0"/>
   <origin xyz="0.75 0 0.2" rpy="0 0 0"/>
   <limit effort="10" velocity="1" lower="-3.14" upper="3.14"/>
    <parent link="link2"/>
    <child link="link3"/>
  </joint>

  <link name="link4">
    <visual>
        <origin xyz="0.18 -0.05 0" rpy="0 3.14 0" />
        <geometry>
          <mesh filename="package://robot_arm/meshes/link4.stl" scale="0.01 0.01 0.01"/>
        </geometry>
         <material name="grey">
        <color rgba="0.83 0.83 0.83 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://robot_arm/meshes/link4.stl" scale="0.01 0.01 0.01"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="10"/>
      <inertia ixx="1e-3" ixy="0.0" ixz="0.0" iyy="1e-3" iyz="0.0" izz="1e-3"/>
    </inertial>
  </link>

   <joint name="link3_to_link4" type="revolute">
   <axis xyz="1 0 0"/>
   <origin xyz="0.85 0 0" rpy="0 0 0"/>
   <limit effort="10" velocity="1" lower="-3.14" upper="3.14"/>
    <parent link="link3"/>
    <child link="link4"/>
  </joint>

  <link name="link5">
    <visual>
        <origin xyz="0 0 0" rpy="-1.57 0 1.57" />
        <geometry>
         <mesh filename="package://robot_arm/meshes/link5.stl" scale="0.01 0.01 0.01"/>
        </geometry>
         <material name="grey">
        <color rgba="0.83 0.83 0.83 1.0"/>
      </material>
    </visual>
    <collision>
      <geometry>
         <mesh filename="package://robot_arm/meshes/link5.stl" scale="0.01 0.01 0.01"/>
      </geometry>
    </collision>
    <inertial>
      <mass value="10"/>
      <inertia ixx="1e-3" ixy="0.0" ixz="0.0" iyy="1e-3" iyz="0.0" izz="1e-3"/>
    </inertial>
  </link>

  <joint name="link4_to_link5_pitch" type="revolute">
   <axis xyz="0 1 0"/>
   <origin xyz="0.3 0 0" rpy="0 0 0"/>
   <limit effort="10" velocity="1" lower="-3.14" upper="3.14"/>
    <parent link="link4"/>
    <child link="link5"/>s
  </joint>

  <link name="link5_roll">
    <inertial>
      <mass value="10"/>
      <inertia ixx="1e-3" ixy="0.0" ixz="0.0" iyy="1e-3" iyz="0.0" izz="1e-3"/>
    </inertial>
  </link>

  <joint name="link4_to_link5_roll" type="revolute">
   <axis xyz="0 0 1"/>
   <origin xyz="0 0 -0.40" rpy="0 0 0"/>
   <limit effort="10" velocity="1" lower="-3.14" upper="3.14"/>
    <parent link="link5"/>
    <child link="link5_roll"/>
  </joint>

  <link name="end_effector">
  </link>

  <joint name="link5_to_end" type="fixed">
   <origin xyz="0 0 0" rpy="0 0 0"/>
    <parent link="link5_roll"/>
    <child link="end_effector"/>
  </joint>

</robot>